### ISSUE TITLE
Drop XHTML support in the new (semantic) HTML converter

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor#4309]
I'm not really sure if XHTML is still relevant or not and if we should support it in the new semantic HTML converter.
According to Dan, it's mainly (only?) used in the Maven Plugin to generate a site since Maven Site only supports XHTML.

Arguably, it can be useful to parse an XHTML document using an XML parser or apply XSLT transformations but in this case I guess using the DocBook 5 converter is probably a better option?

Anyway, if anyone has a strong case to make to keep XHTML support please do so!

**Comment by mojavelinux** [comment]
> According to Dan, it's mainly (only?) used in the Maven Plugin to generate a site since Maven Site only supports XHTML.

I'll be honest, this is the only reason we ever added XHTML support. Though there is a case for being able to use an XML parser to extract data from it since HTML can still be difficult to parse in some languages. I think if you try to remove it, there are people are are going to loudly complain and object. Given it's not terribly difficult to support, I think it should be kept.

**Comment by someth2say** [comment]
If I am allowed to comment, we are one of those people using the XHTML backend.
Our use case is extracting formatted HTML (such as titles) from the document via XPath in order to inject in in other place (something similar to an HTML ToC).
We need the extracted content to be in HTML format, so the DocBook backend is not an (easy) option.

**Comment by mojavelinux** [comment]
Of course you are welcome to comment ;)

Having worked with XML and HTML parsers a fair bit myself, I can attest that the translation from HTML to XML/XHTML is not perfect, so the parsers can actually mess things up. Browsers are a mythical beast in that they can actual get this right, but it comes at a cost. Given that there is very little cost to maintain XHTML-style output (at least as far as the tags go), I think we would be shooting ourselves and community in the foot by removing it.

**Comment by ggrossetie** [comment]
Agreed, keeping XHTML around isn't much of a burden. I don't have a strong use case myself, but the argument about parsing with an XML parser / XPath is a fair one. HTML parsing across languages is often more painful than dealing with well-formed XHTML. If the maintenance cost stays low, I don't see a good reason to drop it.

Let's close this issue!
